### PR TITLE
#221: Split vendor modules bundle by default (closes #221) (Closes #219)

### LINKED ISSUE #219: Add a way to activate BundleAnalyzerPlugin
## requirements
There should be an easy way to activate the bundle analyzer plugin so that the output bundle can be explored easily during development.

## specs
Just conditionally add the following to plugins:
```
new BundleAnalyzerPlugin({ generateStatsFile: true })
```

## misc
{optional: other useful info}

### LINKED ISSUE #221: Split vendor modules bundle by default
## requirements
Splitting vendor modules (i.e., `node_modules`) bundle by default is a good idea since:
1. they change less frequently and can be cached by the client
2. it is easier to have a clearer view of how much of the bundle code comes from dependencies, spurring optimization of the `node_modules` size

## specs
See Webpack docs:
https://webpack.js.org/guides/code-splitting/

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ecamellini**
Closes #221
Closes #219

### tests performed

* Tests work
* See bundle analyzer result

### tests not performed (domain coverage)

> At times not everything can be tested, and writing what hasn't been tested is just as important as writing what has been tested.

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ The default webpack config shipped with scriptoni should be fine in most cases.
 If this is the case, you can override the default config by passing an additional `--webpackConfig` argument, followed by the file path containing your override function.
 
 Let's say, for example, you want to change the output library.
-You can provide a `webpack.config.js` file in the root directory of your project, with the following content:
+You can provide a `webpack.dev.js` file in the root directory of your project, with the following content:
```
Comment (giogonzo) [inline README.md]: why this change? It looks like you only renamed the internal "webpack config for dev" file, while you didn't change the logic to retrieve the user-provided file in any way. This should stay `webpack.config.js` (also because there is a single file both for dev and for build!)

File: README.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ As you can see, your function will receive the default webpack config as first a
 - `NODE_ENV`: 'development' or 'production'
 - `target`: one of `dev`, `build`, `dev-ts` or `build-ts`
 
-As a last step, you can change the `start` and `build` scripts in your `package.json` file by adding `--webpackConfig ./webpack.config.js` and the end of both commands:
+As a last step, you can change the `start` and `build` scripts in your `package.json` file by adding `--webpackConfig ./webpack.dev.js` and the end of both commands:
```
Comment (giogonzo) [inline README.md]: same comment here

File: src/scripts/webpack/webpack.base.js [hunk 3]
```diff
@@ -55,7 +56,9 @@ export default ({ config, paths, NODE_ENV, jsLoader = JSLoader('babel') }) => {
         syntax: 'scss'
       }),
       new HTMLPlugin(getHtmlPluginConfig(NODE_ENV, config, paths))
-    ],
+    ].concat(
+      bundleAnalyzer ? [new BundleAnalyzerPlugin({ generateStatsFile: true })] : []
```
Comment (giogonzo) [inline src/scripts/webpack/webpack.base.js]: why `generateStatsFile: true`?

File: test/webpack.test.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ describe('webpack', () => {
       mkdirSync(resolve(templateDir, 'build'));
       return runCommands([
         `cd ${templateDir}`,
-        'yarn build'
+        'yarn build --bundleAnalyzer'
```
Comment (giogonzo) [inline test/webpack.test.ts]: what does this change for the test? Is it testing the `bundleAnalyzer` setup in some way? It doesn't seem so, I would remove the flag from here

Comment (ecamellini) [reply test/webpack.test.ts]: Leftover

File: src/scripts/webpack/webpack.base.js (hunk 3 above)
Comment (ecamellini) [reply src/scripts/webpack/webpack.base.js]: Some additional analyzer tools require the stats file upload to be used, such as:
* https://chrisbateman.github.io/webpack-visualizer/
* https://webpack.github.io/analyse/
* https://alexkuz.github.io/webpack-chart/

Comment (ecamellini) [reply src/scripts/webpack/webpack.base.js]: Right, let's not speculate, I removed it and we will add it if someone actually needs it in the future 😛
